### PR TITLE
#93 deploy lambda functions from within the tool

### PR DESCRIPTION
**Pull request by DotGav**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -100,8 +101,29 @@ class Main {
...
+      ].data[0];
+      if (provider === 'serverless') {
+        exec(
+          `serverless deploy -f ${selectedLambdaFunctionName} -r ${program.region} --aws-profile ${profile}`,
+          { cwd: location },
+          (error, stdout, stderr) => {
+            if (error) {
+              console.error(`stderr: ${stderr}`);
+              console.error(`exec error: ${error}`);
+            }
```
Comment (antoineeudes) [inline index.js]: shouldn't we log things in the log box? for example `stdout`

File: index.js [hunk 2]
```diff
@@ -100,8 +101,29 @@ class Main {
...
+      ].data[0];
+      return open(
+        `https://${program.region}.console.aws.amazon.com/lambda/home?region=${program.region}#/functions/${program.stackName}-${selectedLambdaFunctionName}?tab=configuration`,
+      );
+    });
+    screen.key(['d', 'D'], () => {
+      const selectedLambdaFunctionName = this.table.rows.items[
+        this.table.rows.selected
+      ].data[0];
+      if (provider === 'serverless') {
```
Comment (BenEllerby) [inline index.js]: maybe change to 'serverlessFramework'

File: index.js (hunk 1 above)
Comment (BenEllerby) [reply index.js]: I think that's best too

File: index.js [hunk 3]
```diff
@@ -100,8 +101,29 @@ class Main {
...
-      return open(`https://${program.region}.console.aws.amazon.com/lambda/home?region=${program.region}#/functions/${selectedLambdaFunctionName}?tab=configuration`);
+      const selectedLambdaFunctionName = this.table.rows.items[
+        this.table.rows.selected
+      ].data[0];
+      return open(
+        `https://${program.region}.console.aws.amazon.com/lambda/home?region=${program.region}#/functions/${program.stackName}-${selectedLambdaFunctionName}?tab=configuration`,
+      );
+    });
+    screen.key(['d', 'D'], () => {
+      const selectedLambdaFunctionName = this.table.rows.items[
```
Comment (BenEllerby) [inline index.js]: can we move the contents of this into its own isolated function?
Would also like to see this unit tested :)

Comment (DotGav) [reply index.js]: Moved to a seperate function testing started on seperate branch, made the decision to move setting up testing out of this branch so a timebox can be taken to setup initial testing

**Pull request merged**